flocker/node/_deploy.py

```diff
@@ -674,6 +690,7 @@ def applications_from_units(units):
                 ))
 
             return [NodeState(
+                uuid=self.uuid,
```

Comment:
And a bunch of other tests on `ApplicationNodeDeployerDiscoverNodeConfigurationTests` cover this case (mostly incidentally, though).